internal/venice-common/src/main/java/com/linkedin/venice/utils/AvroRecordUtils.java

```diff
@@ -84,4 +85,11 @@ public static <T extends SpecificRecord> T prefillAvroRecordWithDefaultValue(T r
 
     return recordType;
   }
+
+  public static void clearRecord(GenericRecord record) {
+    int fieldCount = record.getSchema().getFields().size();
+    for (int i = 0; i < fieldCount; ++i) {
+      record.put(i, null);
```

Comment:
What if one field is primitive type?

Reply:
I borrowed this implementation from `StorageReadRequestsHandler::clearFieldsInReusedRecord`. GenericRecord internally is like a `Map<String, Object>`, so it's correct. It won't work for SpecificRecord however. Btw, I'm adding unit-test to verify that it works as I expect.